distributed/dashboard/scheduler.py

```diff
@@ -268,8 +270,9 @@ def collect(self):
             "Number of tasks known by scheduler.",
             labels=["state"],
         )
-        tasks.add_metric(["received"], len(self.server.tasks))
-        tasks.add_metric(["unrunnable"], len(self.server.unrunnable))
+        task_counter = Counter(map(lambda x: x.state, self.server.tasks.values()))
+        for state, num_tasks in task_counter.items():
+            tasks.add_metric([state], num_tasks)
```

Comment:
This should become simpler if we merge in https://github.com/dask/distributed/pull/3262

Do you have any interest in reviewing that PR?  It would be good to get someone else's eyes on it.

Reply:
In particular, I prefer to avoid looping over all tasks if we can.  It can really slow things down when we have millions of them.

Reply:
Ah, if we're doing this per state anyway then why treat forgotten differently?